packages/zowe-explorer/__tests__/__theia__/theia/extension.theiaFirefox.ts

```diff
@@ -0,0 +1,183 @@
+    await driverFirefox.findElement(By.id(JobsLocators.jobTabId)).click();
+}
+
+export async function clickOnDatasetsPanel(){
+    await driverFirefox.findElement(By.id(DatasetsLocators.datasetsPanelId)).click();
+}
+
+export async function clickOnAddSessionInDatasets(){
+    await driverFirefox.findElement(By.id(DatasetsLocators.datasetsAddSessionId)).click();
+}
+
+export async function clickOnUssPanel(){
+    await driverFirefox.findElement(By.id(UssLocators.ussPanelId)).click();
+}
+
+export async function clickOnAddSessionInUss(){
+    await driverFirefox.findElement(By.id(UssLocators.ussAddSessionId)).click();
+}
+
+export async function clickOnJobsPanel(){
+    await driverFirefox.findElement(By.id (JobsLocators.jobsPanelId)).click();
+}
+
+export async function clickOnAddSessionInJobs(){
+    await driverFirefox.findElement(By.id(JobsLocators.jobsAddSessionId)).click();
+}
+
+export async function addProfileDetails(profileName: string){
+    await driverFirefox.findElement(By.id(DatasetsLocators.datasetsAddSessionId)).click();
+    await driverFirefox.sleep(SHORTSLEEPTIME);
+    await driverFirefox.findElement(By.xpath(DatasetsLocators.emptyInputBoxXpath)).sendKeys(Key.ENTER);
+    await driverFirefox.sleep(SHORTSLEEPTIME);
+    const datasetProfileName = await driverFirefox.wait(until.elementLocated(By.xpath(DatasetsLocators.emptyInputBoxXpath)),WAITTIME);
+    datasetProfileName.sendKeys(profileName);
+    datasetProfileName.sendKeys(Key.ENTER);
+    await driverFirefox.sleep(SHORTSLEEPTIME);
+    const zosUrl = await driverFirefox.findElement(By.xpath(DatasetsLocators.emptyInputBoxXpath));
+    zosUrl.sendKeys("fakehost.net:1003");
+    zosUrl.sendKeys(Key.ENTER);
+    await driverFirefox.sleep(SHORTSLEEPTIME);
+    const username = await driverFirefox.findElement(By.xpath(DatasetsLocators.emptyInputBoxXpath));
+    username.sendKeys(Key.ENTER);
+    await driverFirefox.sleep(SHORTSLEEPTIME);
+    const password = await driverFirefox.findElement(By.xpath(DatasetsLocators.emptyInputBoxXpath));
+    password.sendKeys(Key.ENTER);
+    await driverFirefox.sleep(SHORTSLEEPTIME);
+    const authorization = await driverFirefox.findElement(By.xpath(DatasetsLocators.emptyInputBoxXpath));
+    authorization.sendKeys("False - Accept connections with self-signed certificates");
+    authorization.sendKeys(Key.ENTER);
+    await driverFirefox.sleep(SHORTSLEEPTIME);
+    const basepath = await driverFirefox.findElement(By.xpath(DatasetsLocators.emptyInputBoxXpath));
+    basepath.sendKeys(Key.ENTER);
+    await driverFirefox.sleep(SHORTSLEEPTIME);
+    const encoding = await driverFirefox.findElement(By.xpath(DatasetsLocators.emptyInputBoxXpath));
+    encoding.sendKeys(Key.ENTER);
+    await driverFirefox.sleep(SHORTSLEEPTIME);
+    const responseTimeout = await driverFirefox.findElement(By.xpath(DatasetsLocators.emptyInputBoxXpath));
+    responseTimeout.sendKeys(Key.ENTER);
+    await driverFirefox.sleep(SHORTSLEEPTIME);
```

Comment:
These fields in the add profile dialog are dependent on the zosmf meta yaml file. If the tests are run locally on a different version of zowe cli, the additional/missing fields will cause the tests to fail. Perhaps there is a way the fields that are prompted could come from the meta file itself, rather than specific to the cli version.